.github/workflows/build.yml

```diff
@@ -226,10 +222,24 @@ jobs:
           sudo mv /usr/bin/python3 /usr/bin/python3-default
           sudo ln -s ${{ env.pythonLocation }}/bin/python3 \
           /usr/bin/python3
+      - name: Process user data file
+        id: process-user-data
+        # Process the user data script with the Windows admin password
+        # pulled from AWS Parameter Store. Return the admin password to
+        # pass into Packer as an environment variable.
+        run: |
+          output=$(python .github/scripts/process_user_data_file.py)
+          echo "::add-mask::$output"
+          echo "::set-output name=pass::$output"
```

Comment:
I'd kind of prefer for this to be bundled into the Python script instead of a run of multiple commands. I think that would also consolidate the logic of working with retrieving the password from SSM.

Reply:
@mcdonnnj sorry can you please expand on this? I didn't understand. You want this within the python script? If so `"::add-mask::"` is a gh actions specific command. This was the way I was able to successfully mask the password and it needed to be in this order in multiple commands based on what I found. I tried a few different solutions with no success.

Reply:
for reference: https://github.com/actions/runner/issues/475#issuecomment-635750237

Reply:
@itsmostafa The Action runner processes that command if seen on standard output. The Python script can do:

```python
print(f"::add-mask::{pasword}")
```

to achieve the same effect. I have used the [actions_toolkit](https://github.com/yanglbme/actions-toolkit) package to handle GHA commands in Python before (I'm not suggesting you do that for this) and that's more-or-less how it issues commands:

```python
def issue_command(command: str, properties: dict, message):
    """
    Commands

    Command Format:
        ::name key=value,key=value::message

    Examples:
        ::warning::This is the message
        ::set-env name=MY_VAR::some value
    """
    cmd = Command(command, properties, message)
    sys.stdout.write(str(cmd) + os.linesep)
    ```

Reply:
Oh nice, i didnt know this. that's pretty awesome

Reply:
@mcdonnnj `print(f"::add-mask::{password}")` didnt work for me. refer to [4993b6e](https://github.com/cisagov/windows-server-packer/pull/10/commits/4993b6e4912fa9b85bf532a9f7e8f61d3cebf0e9)

Reply:
reverted with [cc0027a](https://github.com/cisagov/windows-server-packer/pull/10/commits/cc0027a58a0ec6988e4c2a72b19b869a57755305)

Reply:
@itsmostafa You have to do both the `add-mask` and the `set-output` inside the Python script. With appropriate changes in place:

```python
print(f"::add-mask::{password}")
print(f"::set-output name=pass::{password}")
```

The Action should look like this:

```yaml
run: python .github/scripts/process_user_data_file.py
```

I'm sorry if this was unclear from my original comment.